### ISSUE TITLE
On flow.kill(), I get AttributeError: 'NoneType' object has no attribute 'connected' 

### ISSUE THREAD

**Issue by paskyorg** [mitmproxy/mitmproxy#3017]
I want to kill some type of requests, on request event when the flow is not json and is replay.

```
def request(flow):
    print("URL: {}".format(flow.request.path))
    if flow.request.headers.get('Content-Type', None) != "application/json" and flow.request.is_replay:
        flow.kill()
```

The exception is:

> Exception in thread RequestReplayThread:
> Traceback (most recent call last):
>   File "c:\program files (x86)\python36-32\lib\threading.py", line 916, in _bootstrap_inner
>     self.run()
>   File "c:\program files (x86)\python36-32\lib\site-packages\mitmproxy\proxy\protocol\http_replay.py", line 116, in run
>     if server.connected():
> AttributeError: 'NoneType' object has no attribute 'connected'

I'm using mitmproxy 2.0.2 with Python 3.6, on Windows 10.

**Comment by mhils** [comment]
Thanks. It looks like your mitmproxy is outdated. Could you please upgrade an try again?

**Comment by paskyorg** [comment]
Thanks. I will switch to newer version, 3.0.3.

Now my problem with 3.0.3 are the 502 errors (already reported), before never happened with 2.0.2.

**Comment by mhils** [comment]
Closing this as the 502 errors are fixed with 3.0.4 (see #3036).
